hail/python/hail/context.py

```diff
@@ -180,11 +180,34 @@ def init(sc=None, app_name='Hail', master=None, local='local[*]',
          global_seed=6348563392232659379, _backend=None):
     """Initialize Hail and Spark.
 
+    Examples
+    --------
+    Import and initialize Hail using GRCh38 as the default reference genome:
+
+    >>> import hail as hl
+    >>> hl.init(default_reference='GRCh38')  # doctest: +SKIP
+
     Notes
     -----
-    This function is used to initialize Hail and Spark. If Hail functionality
-    that requires initialization is used before :func:`.init` is called, then
-    Hail will be automatically initialized with default arguments.
+    Hail is not only a Python library; most of Hail is written in Java/Scala
+    and runs together with Apache Spark in the Java Virtual Machine (JVM).
+    In order to use Hail, a JVM needs to run as well. The :func:`.init`
+    function is used to initialize Hail and Spark.
+
+    This function also sets global configuration parameters used for the Hail
+    session, like the default reference genome and log file location.
+
+    This function will be called automatically (with default parameters) if
+    any Hail functionality requiring the backend (most of the libary!) is used.
+    To initialize Hail explicitly with non-default arguments, be sure to do so
+    directly after importing the module, as in the above example.
+
+    Note
+    ----
+    If a :class:`pyspark.SparkContext` is already running, then Hail must be
+    initialized with it as an argument:
```

Comment:
Is a similar explanation needed for Backend / SparkBackend?

Reply:
Good question. The answer is no, not right now. During 0.2, we have a backward compatibility mandate: pipelines using documented and non-experimental functionality must continue to work in future versions.

The backend stuff isn't documented right now, because we expect that to change. We don't want to document it right now and lock ourselves into interfaces.